app/commands/v2/unpublish.rb

```diff
@@ -95,68 +93,48 @@ def send_downstream
 
         DownstreamDraftWorker.perform_async_in_queue(
           DownstreamDraftWorker::HIGH_QUEUE,
-          content_id: content_item.content_id,
-          locale: locale,
+          content_id: document.content_id,
+          locale: document.locale,
           payload_version: event.id,
           update_dependencies: true,
         )
 
         DownstreamLiveWorker.perform_async_in_queue(
           DownstreamLiveWorker::HIGH_QUEUE,
-          content_id: content_item.content_id,
-          locale: locale,
+          content_id: document.content_id,
+          locale: document.locale,
           payload_version: event.id,
           update_dependencies: true,
         )
       end
 
-      def locale
-        payload.fetch(:locale, ContentItem::DEFAULT_LOCALE)
-      end
-
       def previous_version_number
         payload[:previous_version].to_i if payload[:previous_version]
       end
 
-      def find_unpublishable_content_item
-        allowed_states = %w(published unpublished)
-
+      def find_unpublishable_edition
         if payload[:allow_draft]
-          allowed_states = %w(draft)
+          edition = document.draft
+        else
+          edition = previous
         end
 
-        content_item = ContentItem.where(
-          content_id: content_id,
-          locale: locale,
-          state: allowed_states
-        ).order(nil).lock.first
-
-        content_item if content_item && (payload[:allow_draft] || !Unpublishing.is_substitute?(content_item))
+        edition if edition && (payload[:allow_draft] || !Unpublishing.is_substitute?(edition))
       end
 
-      def previous_item_should_be_superseded?
-        previous_item && find_unpublishable_content_item != previous_item
+      def previous
+        document.published_or_unpublished
       end
 
-      def previous_item
-        raise "There should only be one previous published or unpublished item" if previous_items.size > 1
-        previous_items.first
+      def previous_edition_should_be_superseded?
+        previous && find_unpublishable_edition != previous
```

Comment:
suggest `previous && (find_unpublishable_edition != previous)` in case there's anyone silly enough not to hold the full set of operator precedence rules in their working memory